app/src/main/java/com/travelguide/fragments/TripPlanListFragment.java

```diff
@@ -40,6 +40,27 @@
     private TextView tvEmpty;
     private RecyclerView rvTripPlans;
 
+    private static String city;
+    private static String group;
+    private static String season;
+
+    public TripPlanListFragment() {
+
+    }
+
+    public static TripPlanListFragment newInstance(String city, String group, String season) {
+        TripPlanListFragment tripPlanListFragment = new TripPlanListFragment();
+        Bundle args = new Bundle();
+        args.putString("city", city);
+        args.putString("group", group);
+        args.putString("season", season);
+        tripPlanListFragment.setArguments(args);
+        TripPlanListFragment.city = city;
```

Comment:
This is wrong, we should not have static fields (it's a problem for Fragment lifecycle). Please remove static ...and see what is done on `TripPlanDetailsFragment` ..that's the right way. Three non-static fields and should read all them inside `onCreate` methods ...like that:

`if (getArguments() != null) {
            mTripPLanObjectId = getArguments().getString(ARG_TRIP_PLAN_OBJECT_ID);
        }`